### ISSUE TITLE
Can we add configable debug settings for delay pod delete when there is a `Error` state of pods ?

### ISSUE THREAD

**Issue by gwind** [apache/airflow#8564]
**Description**

Add configable debug settings for delay pod delete when there is a `Error` state of pods.

**Use case / motivation**

In `apache/airflow:1.10.10` image.

I'm deploy a airflow in k8s, want to use Kubernetes Executor for task excute.
If the pod got Error state, airflow scheduler would delete pod immediately.
So we can not see what happend, pod is deleted in some seconds.

When I add `time.sleep()` in `kubernetes_executor.py:896` , like this:

```python
    def _change_state(self, key, state, pod_id, namespace):
        if state != State.RUNNING:
            if self.kube_config.delete_worker_pods:
                for x in range(120):
                    self.log.info(str(x) + ": sleep 1s for...")
                    time.sleep(1)
                self.kube_scheduler.delete_pod(pod_id, namespace)
                self.log.info('Deleted pod: %s in namespace %s', str(key), str(namespace))
            try:
                self.running.pop(key)
            except KeyError:
                self.log.debug('Could not find key: %s', str(key))
        self.event_buffer[key] = state
```

When trigger execute manully, I can see pod got `Error` state soon.

```
➜  ~ kubectl get po
NAME                                                         READY   STATUS    RESTARTS   AGE
airflow-564c84ff46-tn5mg                                     2/2     Running   0          67s
examplebashoperatorrunme0-76fd68aa96d64e8c93c7c87904f3312a   0/1     Error     0          24s
```

Watch pod's log:

```
➜  ~ kubectl logs -f examplebashoperatorrunme0-76fd68aa96d64e8c93c7c87904f3312a
Traceback (most recent call last):
  File "/home/airflow/.local/bin/airflow", line 23, in <module>
    import argcomplete
ModuleNotFoundError: No module named 'argcomplete'
```

It's a error in container. It's easy to debug now.

**Comment by hcbraun** [comment]
Hi gwind,
were you able to solve the container error?
`ModuleNotFoundError: No module named 'argcomplete'`
I have the same issue in pods with the Kubernetes executor and the example DAGs

There is an option to keep / not delete worker pods:
  AIRFLOW__KUBERNETES__DELETE_WORKER_PODS: "false"

**Comment by gwind** [comment]
> Hi gwind,
> how did you solve the container error?
> `ModuleNotFoundError: No module named 'argcomplete'`

I've solved it by hack the airflow code.

> I have the same issue in pods with the Kubernetes executor and the example DAGs
> 
> There is an option to keep / not delete worker pods:
> AIRFLOW__KUBERNETES__DELETE_WORKER_PODS: "false"

👍 Indeed, there is a option in airflow/config_templates/default_airflow.cfg

Thanks !

**Comment by hcbraun** [comment]
AIRFLOW__KUBERNETES__RUN_AS_USER: "50000"

**Comment by ousatov-ua** [comment]
Hi guys!

How did you solve the problem ?

>> ModuleNotFoundError: No module named 'argcomplete'

is there any setting etc to fix it???

**Comment by gwind** [comment]
> Hi guys!
> 
> How did you solve the problem ?
> 
> > > ModuleNotFoundError: No module named 'argcomplete'
> 
> is there any setting etc to fix it???

This bug caused by wrong user environment in the airflow POD mostly.

You can use `kubectl exec -it ${THE_POD} bash` go to the inside of the airflow POD, then exec airflow command for testing. You would be found that which user is working then.

**Comment by ousatov-ua** [comment]
Hi!
Thanks!

I've setup it as proposed:
AIRFLOW__KUBERNETES__RUN_AS_USER: "50000"

And it did not help.... :((

**Comment by ousatov-ua** [comment]
Can you please post env variables for your airflow-dev-web?

**Comment by ousatov-ua** [comment]
I see airflow-dev-env variables:
AIRFLOW__KUBERNETES__RUN_AS_USER 50000

**Comment by ousatov-ua** [comment]
Ah no! Seems like I have another exception now :)) Dag is not found.

**Comment by ousatov-ua** [comment]
My config is next:

`AIRFLOW__KUBERNETES__GIT_REPO: "https://github.com/repo/airflow-dags.git"
    AIRFLOW__KUBERNETES__GIT_BRANCH: "master"
    AIRFLOW__KUBERNETES__GIT_SYNC_DEPTH: 1
    AIRFLOW__KUBERNETES__GIT_DAGS_FOLDER_MOUNT_POINT: "/opt/airflow/dags"
    AIRFLOW__KUBERNETES__GIT_SYNC_CONTAINER_REPOSITORY: "k8s.gcr.io/git-sync"
    AIRFLOW__KUBERNETES__GIT_SYNC_CONTAINER_TAG: "v3.1.2"
    AIRFLOW__KUBERNETES__GIT_SYNC_INIT_CONTAINER_NAME: "git-sync-clone"
    #AIRFLOW__KUBERNETES__GIT_SYNC_ONE_TIME: "True"

    #AIRFLOW__KUBERNETES__DAGS_VOLUME_SUBPATH
    AIRFLOW__KUBERNETES__DAGS_VOLUME_MOUNT_POINT: "/opt/airflow/dags"

    AIRFLOW__KUBERNETES__GIT_SYNC_RUN_AS_USER: 0`

When I run dag I have executed git-sync container on worker pod, after that worker tries to load dag by path /opt/airflow/dags and cannot find it by id.... Everything work ok when I use loaded examples:
AIRFLOW__CORE__LOAD_EXAMPLES: "True"

**Comment by ashb** [comment]
This has already been added to master via https://github.com/apache/airflow/pull/7507 (and then renamed in https://github.com/apache/airflow/pull/8312/) -- we're hoping to include this in the 1.10.11 release.

**Comment by kaxil** [comment]
Looks like the issues was fixed by  #7507 (and then renamed in #8312)
